src/vmstats_server.erl

```diff
@@ -194,26 +195,80 @@ system_stats(BaseKey, SystemStats) ->
         _Else ->
             % system load
             SystemStats2 = system_stats:proc_loadavg(SystemStats),
-            statsderl:gauge([BaseKey, <<"system.load_1">>], SystemStats2#stats.load_1, 1.00),
-            statsderl:gauge([BaseKey, <<"system.load_5">>], SystemStats2#stats.load_5, 1.00),
-            statsderl:gauge([BaseKey, <<"system.load_15">>], SystemStats2#stats.load_15, 1.00),
+            log_gauge([BaseKey, <<"system.load_1">>], SystemStats2#stats.load_1, 1.00),
+            log_gauge([BaseKey, <<"system.load_5">>], SystemStats2#stats.load_5, 1.00),
+            log_gauge([BaseKey, <<"system.load_15">>], SystemStats2#stats.load_15, 1.00),
 
             % system cpu %
             SystemStats3 = system_stats:proc_pidstat(os:getpid(), SystemStats2),
             SystemStats4 = system_stats:proc_stat(SystemStats3),
             {CpuUser, CpuSystem} = system_stats_utils:cpu_percent(SystemStats, SystemStats4),
             CpuPercent = trunc(SystemStats4#stats.cpu_cores * (CpuUser + CpuSystem)),
-            statsderl:gauge([BaseKey, <<"system.cpu_percent">>], CpuPercent, 1.00),
+            log_gauge([BaseKey, <<"system.cpu_percent">>], CpuPercent, 1.00),
 
             % system memory
             Vsize = trunc(bytes_to_megabytes(SystemStats4#stats.mem_vsize)),
             Rss = trunc(bytes_to_megabytes(?PAGE_SIZE * (SystemStats4#stats.mem_rss))),
-            statsderl:gauge([BaseKey, <<"system.vsize">>], Vsize, 1.00),
-            statsderl:gauge([BaseKey, <<"system.rss">>], Rss, 1.00),
+            log_gauge([BaseKey, <<"system.vsize">>], Vsize, 1.00),
+            log_gauge([BaseKey, <<"system.rss">>], Rss, 1.00),
 
             SystemStats4
     end.
 
 unix_tstamp_ms() ->
     {Mega, Sec, Micro} = os:timestamp(),
     (Mega * 1000000000 + Sec * 1000) + trunc(Micro / 1000).
+
+dynamic_log_gauge(Key, Val, Rate) ->
+    statsderl:gauge(Key, Val, Rate),
+    erl_optics:gauge_set_alloc(list_to_binary(Key), Val).
+
+log_gauge(Key, Val, Rate) ->
+    statsderl:gauge(Key, Val, Rate),
+    erl_optics:gauge_set(list_to_binary(Key), Val).
+
+log_counter(Key, Val, Rate) ->
+    statsderl:increment(Key, Val, Rate),
+    erl_optics:counter_inc(list_to_binary(Key), Val).
+
+log_timing(Key, Val, Rate) ->
+    statsderl:timing_now(Key, Val, Rate),
+    erl_optics:dist_record_timing_now(list_to_binary(Key), Val).
+
+get_optics_lenses() ->
+    BaseKey = case application:get_env(vmstats, base_key) of
+        {ok, Key} -> <<Key/binary, $.>>;
+        undefined -> <<"">>
+    end,
```

Comment:
I believe you copied this from the init function.  A better approach would be to refactor into a function so that we can guarantee that the results are always in sync.